jib-core/src/test/java/com/google/cloud/tools/jib/tar/TarExtractorTest.java

```diff
@@ -79,4 +83,53 @@ public void testExtract_symlinks() throws URISyntaxException, IOException {
     Assert.assertTrue(Files.isSymbolicLink(destination.resolve("directory-symlink")));
     Assert.assertTrue(Files.isSymbolicLink(destination.resolve("directory1/file-symlink")));
   }
+
+  @Test
+  public void testExtract_modificationTimePreserved() throws URISyntaxException, IOException {
+    Path source = Paths.get(Resources.getResource("core/extract.tar").toURI());
+    Path destination = temporaryFolder.getRoot().toPath();
+
+    TarExtractor.extract(source, destination);
+
+    assertThat(Files.getLastModifiedTime(destination.resolve("file A")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:13:09Z")));
+    assertThat(Files.getLastModifiedTime(destination.resolve("file B")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:12:00Z")));
+    assertThat(Files.getLastModifiedTime(destination.resolve("folder")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:12:33Z")));
+    assertThat(Files.getLastModifiedTime(destination.resolve("folder").resolve("nested folder")))
+        .isEqualTo(FileTime.from(Instant.parse("2019-08-01T16:13:30Z")));
+    assertThat(
+            Files.getLastModifiedTime(
+                destination.resolve("folder").resolve("nested folder").resolve("file C")))
```

Comment:
Just FYI, no need to change: in *tests*, we're tolerant in writing `destination.resolve("folder/nested folder/file C"`) if it can simplify code or improve readability.